src/openrct2/ride/RideConstruction.cpp

```diff
@@ -1317,13 +1315,17 @@ CoordsXYZD ride_get_entrance_or_exit_position_from_screen_position(const ScreenC
             entranceExitCoords.direction = mapX < 0 ? 0 : 2;
         }
 
-        for (int32_t i = 0; i < MAX_STATIONS; i++)
```

Comment:
It's a coincidence that there are 4 stations max and 4 directions

Reply:
good catch. Should perhaps change the variable name or the constant name to make it even more clear.
I would perhaps even redesign the function so that its clear what its iterating on:
```
for (Direction checkDirection = 0; checkDirection  < 4; checkDirection++)
{
    entranceExitCoords.direction = (entranceExitCoords.direction + checkDirection) & 0x3;
    /// continue the rest of the function...
    // search for TrackElement one tile over, shifted in the search direction
   auto nextLocation = entranceExitCoords;
   /// remember to remove L1365